### PR TITLE
Move examples to a new repository

### PR DESCRIPTION
**Pull request by toshihikoyanase**
## Motivation

This PR is related to #2654.

## Description of the changes

- Remove examples. They will be placed in a new repository by using `git subtree split` command. The prototype repository is https://github.com/toshihikoyanase/optuna-examples.

### TODO

- [x] Update README.md
- [x] Update links in the references

### Opinion Wanted

- Do we move all examples? Or do we keep some basic examples like `quadratic_simple.py`?

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -9,63 +9,19 @@ This page contains a list of example codes written with Optuna.
...
-* [RAPIDS](./rapids_simple.py)
-* [Scikit-learn](./sklearn/sklearn_simple.py)
-* [Scikit-learn OptunaSearchCV](./sklearn/sklearn_optuna_search_cv_simple.py)
-* [Scikit-image](./skimage_lbp_simple.py)
-* [SKORCH](./pytorch/skorch_simple.py)
-* [Tensorflow](./tensorflow/tensorflow_estimator_simple.py)
-* [Tensorflow (eager)](./tensorflow/tensorflow_eager_simple.py)
-* [XGBoost](./xgboost/xgboost_simple.py)
-
+The examples with ML libraries was moved to [optuna/optuna-examples](https://github.com/optuna/optuna-examples/).
```
[the remark this answers is not in the document]
Comment (hvy) [reply examples/README.md:12]: From my experience, these URLs tend to break (silently) from time to time. I think it's enough to keep a single URL to the example repository.

File: README.md [hunk 2]
```diff
@@ -85,20 +85,20 @@ study.optimize(objective, n_trials=100)  # Invoke optimization of the objective
...
+* FastAI ([V1](https://github.com/optuna/optuna-examples/tree/main/fastai/fastaiv1_simple.py), [V2](https://github.com/optuna/optuna-examples/tree/main/fastai/fastaiv2_simple.py))
+* [Keras](https://github.com/optuna/optuna-examples/tree/main/keras/keras_integration.py)
+* [LightGBM](https://github.com/optuna/optuna-examples/tree/main/lightgbm/lightgbm_integration.py)
+* [MXNet](https://github.com/optuna/optuna-examples/tree/main/mxnet/mxnet_integration.py)
+* [PyTorch](https://github.com/optuna/optuna-examples/tree/main/pytorch/pytorch_simple.py)
+* [PyTorch Ignite](https://github.com/optuna/optuna-examples/tree/main/pytorch/pytorch_ignite_simple.py)
+* [PyTorch Lightning](https://github.com/optuna/optuna-examples/tree/main/pytorch/pytorch_lightning_simple.py)
+* [TensorFlow](https://github.com/optuna/optuna-examples/tree/main/tensorflow/tensorflow_estimator_integration.py)
+* [tf.keras](https://github.com/optuna/optuna-examples/tree/main/tfkeras/tfkeras_integration.py)
+* [XGBoost](https://github.com/optuna/optuna-examples/tree/main/xgboost/xgboost_integration.py)
```
[the remark this answers is not in the document]
Comment (toshihikoyanase) [reply README.md:101]: > we can reference to Optuna's API reference
Sounds reasonable. I'll create a follow-up PR to update the links.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! I agree with removing these examples from `optuna.optuna`. Could you resolve the conflict?

**Comment by toshihikoyanase** [comment]
@HideakiImamura Thank you for letting me know. I resolved the conflict. PTAL.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update! LGTM!

**Review by hvy — APPROVED** [review]
LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM again!

**Pull request merged**